File: quickSort.py
def quickSort(L, low, high):
    i = low
    j = high
    if(i >= j):
        return L
    key = L[i]
    while(i < j):
        #从右向左寻找是否有比Key小的数
        while i < j and L[j] >= key:
            j = j - 1
        L[i] = L[j]
        #从左向右寻找是否有比Key大的数
        while i < j and L[i] <= key:
            i = i + 1 
        L[j] = L[i]
    L[i] = key
    quickSort(L, low, i-1)
    quickSort(L, i+1, high)
    return L

File: quickSort.c
#include<stdio.h>

void quickSort(int *a, int l, int r){
    if(l >= r){
        return;
    }
    int i = l;
    int j = r;
    int key = a[l];

    while(i < j){
        while(i < j && key <= a[j]){
            j--;
        }
        a[i] = a[j];
        
        while(i < j && key >= a[i]){
            i++;
        }
        a[j] = a[i];
    }

    a[i] = key;
    quickSort(a, l, i-1);
    quickSort(a, i+1, r);    
}

int main(){
    int i;
    int a[]={5, 6, 4, 1, 3};
    quickSort(a, 0, 4);
    for(i=0; i<=4; i++){
        printf("%d\n", a[i]);
    }
}
